topic/machine-learning/llm-langchain/cratedb_rag_customer_support_langchain.ipynb

```diff
@@ -320,23 +353,72 @@
+      "File \u001b[0;32m/opt/homebrew/lib/python3.10/site-packages/crate/client/http.py:228\u001b[0m, in \u001b[0;36m_raise_for_status_real\u001b[0;34m(response)\u001b[0m\n\u001b[1;32m    226\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m ProgrammingError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m.\u001b[39mjoin(errors))\n\u001b[1;32m    227\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(error, \u001b[38;5;28mdict\u001b[39m):\n\u001b[0;32m--> 228\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m ProgrammingError(error\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmessage\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m),\n\u001b[1;32m    229\u001b[0m                                error_trace\u001b[38;5;241m=\u001b[39merror_trace)\n\u001b[1;32m    230\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m ProgrammingError(error, error_trace\u001b[38;5;241m=\u001b[39merror_trace)\n\u001b[1;32m    231\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m ProgrammingError(message)\n",
+      "\u001b[0;31mProgrammingError\u001b[0m: (crate.client.exceptions.ProgrammingError) SQLParseException[vector query dimension: 1536 differs from field dimension: 768]\n[SQL: SELECT embedding.collection_id AS embedding_collection_id, embedding.embedding AS embedding_embedding, embedding.document AS embedding_document, embedding.cmetadata AS embedding_cmetadata, embedding.custom_id AS embedding_custom_id, embedding.uuid AS embedding_uuid, embedding._score AS \"_score\" \nFROM embedding JOIN collection ON embedding.collection_id = collection.uuid \nWHERE embedding.collection_id IN (?) AND knn_match(embedding.embedding, ?, ?) ORDER BY \"_score\" DESC \n LIMIT ?]\n[parameters: ('38b1b477-c407-43c1-b448-5c47dd0677ed', [0.00435982478381371, -0.0070164900711971205, -0.017386690583236015, -0.009653428082896091, -0.034457741030230506, 0.011580167871809374, -0.0488458186 ... (34028 characters truncated) ... 677578292163, -0.013322782385269863, -0.005240995678386175, -0.014730025761513913, -0.01454590048606112, -0.004257898042583784, -0.010015102564799323], 4, 4)]\n(Background on this error at: https://sqlalche.me/e/20/f405)"
+     ]
+    }
```

Comment:
Loads of traceback information here now. Is it intended? Can it be deleted?

Reply:
Another solution will be to delete everything in CrateDB `embedding` table before writing Jira AI embeddings.

Reply:
I think one of those options should be applied here, to avoid corresponding confusions, right.

Reply:
Thank you. I think it will be good to merge.

Please make sure to squash all your commits into a single one in this case, because individual ones are noisy, also partly removing noise introduced by previous ones, so they don't have much value to be forwarded into mainline individually. Thanks!